vertical-pod-autoscaler/e2e/v1/actuation.go

```diff
@@ -278,6 +287,51 @@ var _ = ActuationSuiteE2eDescribe("Actuation", func() {
 		ginkgo.By(fmt.Sprintf("Waiting for pods to be evicted, hoping it won't happen, sleep for %s", VpaEvictionTimeout.String()))
 		CheckNoPodsEvicted(f, MakePodSet(podList))
 	})
+
+	ginkgo.It("omits injected sidecars", func() {
+		const (
+			servicePort   = int32(8443)
+			containerPort = int32(8444)
+		)
+		// Webhook setup.
+		client := f.ClientSet
+		namespaceName := f.Namespace.Name
+		defer utils.CleanWebhookTest(client, namespaceName)
+
+		// Make sure the namespace created for the test is labeled to be selected by the webhooks.
+		utils.LabelNamespace(f, f.Namespace.Name)
+		utils.CreateWebhookConfigurationReadyNamespace(f)
+
+		ginkgo.By("Setting up server cert")
+		context := utils.SetupServerCert(namespaceName, utils.ServiceName)
+		utils.CreateAuthReaderRoleBinding(f, namespaceName)
+
+		utils.DeployWebhookAndService(f, agnhostImage, context, servicePort, containerPort, sidecarParam)
+
+		// Webhook must be placed after vpa webhook. Webhooks are registered alphabetically.
+		// Use name that starts with "z".
+		webhookCleanup := utils.RegisterMutatingWebhookForPod(f, "z-sidecar-injection-webhook", context, servicePort)
+		defer webhookCleanup()
+
+		// Actual test.
+		SetupVPA(f, "100m", vpa_types.UpdateModeInitial, hamsterTargetRef)
+		SetupHamsterDeployment(f, "100m", "100Mi", defaultHamsterReplicas)
+
+		podList, err := GetHamsterPods(f)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, pod := range podList.Items {
+			observedContainers, ok := pod.GetAnnotations()[annotations.VpaObservedContainersLabel]
+			gomega.Expect(ok).To(gomega.Equal(true))
+			containers, err := annotations.ParseVpaObservedContainersValue(observedContainers)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			gomega.Expect(containers).To(gomega.HaveLen(1))
+			gomega.Expect(pod.Spec.Containers).To(gomega.HaveLen(2))
+		}
+
+		podSet := MakePodSet(podList)
+		ginkgo.By(fmt.Sprintf("Waiting for pods to be evicted, hoping it won't happen, sleep for %s", VpaEvictionTimeout.String()))
+		CheckNoPodsEvicted(f, podSet)
```

Comment:
This second part sounds like a basic eviction test that already is covered in e2e actuation tests...?

Reply:
yeah, I know, but this comment was prompted by the fact that if I did not point out that you have Initial mode in your VPA object, the test would pass but the sidecar detection would not be working

Reply:
I want to keep test simple - it should check just one thing.
IMHO the test logic/correctness should be verified during the pr review. Otherwise this will result with tests for tests...